crud/select/compat/select_old.lua

```diff
@@ -118,7 +118,8 @@ local function build_select_iterator(space_name, user_conditions, opts)
     -- set replicasets to select from
     local replicasets_to_select = replicasets
 
-    if plan.sharding_key ~= nil and opts.force_map_call ~= true then
+    local bucket_id_is_known = plan.sharding_key ~= nil or opts.bucket_id ~= nil
+    if bucket_id_is_known and opts.force_map_call ~= true then
```

Comment:
`bucket_id_is_known = plan.sharding_key ~= nil ...`
I think that `plan.sharding_key ~= nil` doesn't guarantee that bucket id is known. I would calculate a bucket_id and then: `bucket_id_is_known = bucket_id ~= nil or opts.bucket_id ~= nil`. What do you think?

```lua
    local bucket_id                                                                 
    if plan.sharding_key ~= nil then                                                
        bucket_id = sharding.key_get_bucket_id(plan.sharding_key, opts.bucket_id)
    end                                                                             
                                                                                    
    local bucket_id_is_known = bucket_id ~= nil or opts.bucket_id ~= nil            
    if bucket_id_is_known and opts.force_map_call ~= true then                      
        local err                                                                   
        replicasets_to_select, err = common.get_replicasets_by_sharding_key(bucket_id)
        if err ~= nil then                                                          
            return nil, err, true                                                   
        end                                                                      
    end 
```

Reply:
> It is subtle difference: if we know a sharding key, we know a bucket id in fact (just by applying the sharding function).

What will happen if a function for bucket id calculation return not a number?

Reply:
I stopped on the latest option now. Please, check it out.

Reply:
(I didn't try to brute force you. Just a bit puzzled how to better structurize this code.)